[CI] GroupByOptimizerTests testOrderByScriptAndType failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/rgo44odoshacu/tests/:x-pack:plugin:transform:test/org.elasticsearch.xpack.transform.transforms.pivot.GroupByOptimizerTests/testOrderByScriptAndType

**Reproduction line:**
`./gradlew ':x-pack:plugin:transform:test' --tests "org.elasticsearch.xpack.transform.transforms.pivot.GroupByOptimizerTests.testOrderByScriptAndType" -Dtests.seed=FA72B0E58B148A04 -Dtests.locale=uk-UA -Dtests.timezone=Brazil/DeNoronha -Druntime.java=11`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.transform.transforms.pivot.GroupByOptimizerTests&tests.test=testOrderByScriptAndType

**Failure excerpt:**
```
java.lang.AssertionError: expected:<[date3, date2, date1, terms1, terms2]> but was:<[date2, date3, date1, terms1, terms2]>

  at __randomizedtesting.SeedInfo.seed([FA72B0E58B148A04:D37ACA340EA6203D]:0)
  at org.junit.Assert.fail(Assert.java:88)
  at org.junit.Assert.failNotEquals(Assert.java:834)
  at org.junit.Assert.assertEquals(Assert.java:118)
  at org.junit.Assert.assertEquals(Assert.java:144)
  at org.elasticsearch.xpack.transform.transforms.pivot.GroupByOptimizerTests.testOrderByScriptAndType(GroupByOptimizerTests.java:96)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:566)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:829)

```

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/pivot/DateHistogramGroupSourceTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/pivot/DateHistogramGroupSourceTests.java
@@ -34,20 +34,28 @@ public class DateHistogramGroupSourceTests extends AbstractSerializingTestCase<D
         return randomDateHistogramGroupSource(Version.CURRENT, false);
     }
 
+    public static DateHistogramGroupSource randomDateHistogramGroupSourceNoScript(String fieldPrefix) {
+        return randomDateHistogramGroupSource(Version.CURRENT, false, fieldPrefix);
+    }
+
     public static DateHistogramGroupSource randomDateHistogramGroupSource(Version version) {
         return randomDateHistogramGroupSource(version, randomBoolean());
     }
 
     public static DateHistogramGroupSource randomDateHistogramGroupSource(Version version, boolean withScript) {
+        return randomDateHistogramGroupSource(version, withScript, "");
+    }
+
+    public static DateHistogramGroupSource randomDateHistogramGroupSource(Version version, boolean withScript, String fieldPrefix) {
         ScriptConfig scriptConfig = null;
         String field;
 
         // either a field or a script must be specified, it's possible to have both, but disallowed to have none
         if (version.onOrAfter(Version.V_7_7_0) && withScript) {
             scriptConfig = ScriptConfigTests.randomScriptConfig();
-            field = randomBoolean() ? null : randomAlphaOfLengthBetween(1, 20);
+            field = randomBoolean() ? null : fieldPrefix + randomAlphaOfLengthBetween(1, 20);
         } else {
-            field = randomAlphaOfLengthBetween(1, 20);
+            field = fieldPrefix + randomAlphaOfLengthBetween(1, 20);
         }
         boolean missingBucket = version.onOrAfter(Version.V_7_10_0) ? randomBoolean() : false;
 

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/pivot/HistogramGroupSourceTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/pivot/HistogramGroupSourceTests.java
@@ -24,20 +24,28 @@ public class HistogramGroupSourceTests extends AbstractSerializingTestCase<Histo
         return randomHistogramGroupSource(Version.CURRENT, false);
     }
 
+    public static HistogramGroupSource randomHistogramGroupSourceNoScript(String fieldPrefix) {
+        return randomHistogramGroupSource(Version.CURRENT, false, fieldPrefix);
+    }
+
     public static HistogramGroupSource randomHistogramGroupSource(Version version) {
         return randomHistogramGroupSource(version, randomBoolean());
     }
 
     public static HistogramGroupSource randomHistogramGroupSource(Version version, boolean withScript) {
+        return randomHistogramGroupSource(version, withScript, "");
+    }
+
+    public static HistogramGroupSource randomHistogramGroupSource(Version version, boolean withScript, String fieldPrefix) {
         ScriptConfig scriptConfig = null;
         String field;
 
         // either a field or a script must be specified, it's possible to have both, but disallowed to have none
         if (version.onOrAfter(Version.V_7_7_0) && withScript) {
             scriptConfig = ScriptConfigTests.randomScriptConfig();
-            field = randomBoolean() ? null : randomAlphaOfLengthBetween(1, 20);
+            field = randomBoolean() ? null : fieldPrefix + randomAlphaOfLengthBetween(1, 20);
         } else {
-            field = randomAlphaOfLengthBetween(1, 20);
+            field = fieldPrefix + randomAlphaOfLengthBetween(1, 20);
         }
 
         boolean missingBucket = version.onOrAfter(Version.V_7_10_0) ? randomBoolean() : false;

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/pivot/TermsGroupSourceTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/pivot/TermsGroupSourceTests.java
@@ -31,20 +31,28 @@ public class TermsGroupSourceTests extends AbstractSerializingTestCase<TermsGrou
         return randomTermsGroupSource(Version.CURRENT, false);
     }
 
+    public static TermsGroupSource randomTermsGroupSourceNoScript(String fieldPrefix) {
+        return randomTermsGroupSource(Version.CURRENT, false, fieldPrefix);
+    }
+
     public static TermsGroupSource randomTermsGroupSource(Version version) {
         return randomTermsGroupSource(Version.CURRENT, randomBoolean());
     }
 
     public static TermsGroupSource randomTermsGroupSource(Version version, boolean withScript) {
+        return randomTermsGroupSource(Version.CURRENT, withScript, "");
+    }
+
+    public static TermsGroupSource randomTermsGroupSource(Version version, boolean withScript, String fieldPrefix) {
         ScriptConfig scriptConfig = null;
         String field;
 
         // either a field or a script must be specified, it's possible to have both, but disallowed to have none
         if (version.onOrAfter(Version.V_7_7_0) && withScript) {
             scriptConfig = ScriptConfigTests.randomScriptConfig();
-            field = randomBoolean() ? null : randomAlphaOfLengthBetween(1, 20);
+            field = randomBoolean() ? null : fieldPrefix + randomAlphaOfLengthBetween(1, 20);
         } else {
-            field = randomAlphaOfLengthBetween(1, 20);
+            field = fieldPrefix + randomAlphaOfLengthBetween(1, 20);
         }
 
         boolean missingBucket = version.onOrAfter(Version.V_7_10_0) ? randomBoolean() : false;

--- a/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/pivot/GroupByOptimizerTests.java
+++ b/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/pivot/GroupByOptimizerTests.java
@@ -75,12 +75,12 @@ public class GroupByOptimizerTests extends ESTestCase {
     public void testOrderByScriptAndType() {
         Map<String, SingleGroupSource> groups = new LinkedHashMap<>();
 
-        groups.put("terms1", randomTermsGroupSourceNoScript());
+        groups.put("terms1", randomTermsGroupSourceNoScript("t1"));
         // create with scripts
-        groups.put("date1", randomDateHistogramGroupSource(Version.CURRENT, true));
-        groups.put("terms2", randomTermsGroupSource(Version.CURRENT, true));
-        groups.put("date2", randomDateHistogramGroupSourceNoScript());
-        groups.put("date3", randomDateHistogramGroupSourceNoScript());
+        groups.put("date1", randomDateHistogramGroupSource(Version.CURRENT, true, "d1"));
+        groups.put("terms2", randomTermsGroupSource(Version.CURRENT, true, "t2"));
+        groups.put("date2", randomDateHistogramGroupSourceNoScript("d2"));
+        groups.put("date3", randomDateHistogramGroupSourceNoScript("d3"));
 
         List<String> groupNames = GroupByOptimizer.reorderGroups(Collections.unmodifiableMap(groups), Collections.emptySet())
             .stream()

